### PR TITLE
cleanup pkg/util packages

### PR DESCRIPTION
**Pull request by lburgazzoli**
**Release Note**
```release-note
NONE
```

### CODE REVIEW

File: pkg/util/maven/maven_command.go [hunk 1]
```diff
@@ -185,7 +185,7 @@ func (c *Context) AddSystemProperty(name string, value string) {
 	c.AddArgumentf("-D%s=%s", name, value)
 }
 
-func generateProjectStructure(context Context, project Project) error {
+func generateProjectStructure(context Context, project *Project) error {
```
Comment (squakez) [inline pkg/util/maven/maven_command.go:188]: Out of curiosity, what's the reason of this change? what's the gain? I understand we're going to create a new Project struct in this way, right?

**Review by squakez — APPROVED** [review]

**Comment by squakez** [comment]
Feel free to add `trigger native test` label when there are possible impact to Quarkus native (ie, when we build and app).

File: pkg/util/maven/maven_command.go (hunk 1 above)
Comment (lburgazzoli) [reply pkg/util/maven/maven_command.go]: The opposite, this way we are passing a pointer so no copy is performed when invoking this method but I reverted it as it is enough to pass by value in this case. It was a leftover of a change to make the methods on the `Project` struct consistently using pointer receiver instead of a mix of pointer and value receivers.

Comment (squakez) [reply pkg/util/maven/maven_command.go]: Yes, indeed. I think I inverted the red and the green while reading.

**Comment by lburgazzoli** [comment]
The failing test is marked a problematic and fails for a slightly different metric values:

https://github.com/apache/camel-k/blob/4114271b6ccbc74e52a87455123fb522bbbf84a1/e2e/advanced/operator_metrics_test.go#L48-L55

I've created a follow up issue https://github.com/apache/camel-k/issues/5483

**Comment by squakez** [comment]
> The failing test is marked a problematic and fails for a slightly different metric values:
> 
> https://github.com/apache/camel-k/blob/4114271b6ccbc74e52a87455123fb522bbbf84a1/e2e/advanced/operator_metrics_test.go#L48-L55
> 
> I've created a follow up issue #5483

Please, don't rely on those marked as problematic. For some reason they were marked like that in order to identify certain problems happening on Openshift only. In github actions they should run correctly. If some failure is happening it could really because of the changes in the PR, given that we're changing the way to log build output. I've restarted the check, if it clears, then feel free to merge. If not, please, have a look at how the code change is affecting the expected behavior. Thanks.

**Comment by squakez** [comment]
@lburgazzoli yes, if in the overall executions the failures were different, feel free to merge. We have the flaky test tracked for a fix.

**Comment by lburgazzoli** [comment]
> @lburgazzoli yes, if in the overall executions the failures were different, feel free to merge. We have the flaky test tracked for a fix.

https://github.com/apache/camel-k/issues/5485
